components/booking/TimeOptions.tsx

```diff
@@ -1,48 +1,56 @@
 // TODO: replace headlessui with radix-ui
 import { Switch } from "@headlessui/react";
+import { useRouter } from "next/router";
 import { FC, useEffect, useState } from "react";
 import TimezoneSelect, { ITimezoneOption } from "react-timezone-select";
 
 import classNames from "@lib/classNames";
+import { timeZone } from "@lib/clock";
 import { useLocale } from "@lib/hooks/useLocale";
-
-import { is24h, timeZone } from "../../lib/clock";
+import { useToggleQuery } from "@lib/hooks/useToggleQuery";
 
 type Props = {
   localeProp: string;
   onSelectTimeZone: (selectedTimeZone: string) => void;
-  onToggle24hClock: (is24hClock: boolean) => void;
 };
 
 const TimeOptions: FC<Props> = (props) => {
   const [selectedTimeZone, setSelectedTimeZone] = useState("");
-  const [is24hClock, setIs24hClock] = useState(false);
+  const [is24hClock, setIs24hClock] = useState<boolean | null>(null);
```

Comment:
```suggestion
  const [is24hClock, setIs24hClock] = useState(false);
```

There's no need for `null` here, is being treated as false anyways. Unless we're strictly checking for `null`.